src/expr/benches/expr.rs

```diff
@@ -0,0 +1,183 @@
+    });
+    c.bench_function("expr/eq/i32", |bencher| {
+        let mul = new_binary_expr(Type::Equal, DataType::Int32, i0(), i1()).unwrap();
+        bencher.iter(|| mul.eval(&input).unwrap())
+    });
+    c.bench_function("expr/sum/i32", |bencher| {
+        let mut sum =
+            create_agg_state_unary(DataType::Int32, 0, AggKind::Sum, DataType::Int64, false)
+                .unwrap();
+        bencher.iter(|| sum.update_multi(&input, 0, 1024).unwrap())
+    });
+}
+
+/// Evaluate on raw Rust array.
+///
+/// This could be used as a baseline to compare and tune our expressions.
+fn bench_raw(c: &mut Criterion) {
+    // ~55ns
+    c.bench_function("raw/sum/i32", |bencher| {
+        let a = (0..CHUNK_SIZE as i32).collect::<Vec<_>>();
+        bencher.iter(|| a.iter().sum::<i32>())
+    });
+    // ~90ns
+    c.bench_function("raw/add/i32", |bencher| {
+        let a = (0..CHUNK_SIZE as i32).collect::<Vec<_>>();
+        let b = (0..CHUNK_SIZE as i32).collect::<Vec<_>>();
+        bencher.iter(|| {
+            a.iter()
+                .zip(b.iter())
+                .map(|(a, b)| a + b)
+                .collect::<Vec<_>>()
+        })
+    });
+    // ~600ns
+    c.bench_function("raw/add/i32/zip_eq", |bencher| {
+        let a = (0..CHUNK_SIZE as i32).collect::<Vec<_>>();
+        let b = (0..CHUNK_SIZE as i32).collect::<Vec<_>>();
+        bencher.iter(|| {
+            itertools::Itertools::zip_eq(a.iter(), b.iter())
+                .map(|(a, b)| a + b)
+                .collect::<Vec<_>>()
+        })
+    });
+    // ~950ns
+    c.bench_function("raw/add/Option<i32>/zip_eq", |bencher| {
+        let a = (0..CHUNK_SIZE as i32).map(Some).collect::<Vec<_>>();
+        let b = (0..CHUNK_SIZE as i32).map(Some).collect::<Vec<_>>();
+        bencher.iter(|| {
+            itertools::Itertools::zip_eq(a.iter(), b.iter())
+                .map(|(a, b)| match (a, b) {
+                    (Some(a), Some(b)) => Some(a + b),
+                    _ => None,
+                })
+                .collect::<Vec<_>>()
+        })
+    });
+    // ~2100ns
+    c.bench_function("raw/add/Option<i32>/zip_eq,checked", |bencher| {
+        let a = (0..CHUNK_SIZE as i32).map(Some).collect::<Vec<_>>();
+        let b = (0..CHUNK_SIZE as i32).map(Some).collect::<Vec<_>>();
+        struct Overflow;
+        bencher.iter(|| {
+            use itertools::Itertools;
+            itertools::Itertools::zip_eq(a.iter(), b.iter())
+                .map(|(a, b)| match (a, b) {
+                    (Some(a), Some(b)) => a.checked_add(*b).ok_or(Overflow).map(Some),
+                    _ => Ok(None),
+                })
+                .try_collect::<_, Vec<_>, Overflow>()
+        })
+    });
+    // ~2400ns
+    c.bench_function("raw/add/Option<i32>/zip_eq,checked,cast", |bencher| {
+        let a = (0..CHUNK_SIZE as i32).map(Some).collect::<Vec<_>>();
+        let b = (0..CHUNK_SIZE as i32).map(Some).collect::<Vec<_>>();
```

Comment:
We should put these into sth like preparation?

Reply:
They are now. Only the code inside `bencher.iter(...)` is benchmarked.